### PR TITLE
chore(IT Wallet): [SIW-1804] Already activated wallet screen

### PR DESCRIPTION
**Pull request by ale-mazz**
## Short description
This PR adds a new error screen visible if the user tries to activate the wallet when it is already active, starting with the CTA of the activation message

## List of changes proposed in this pull request
- Added a new `ItwAlreadyActiveScreen`
- Mapped the navigation inside `useItwLinkingOptions`

## How to test
From the message CTA with an active eID try to navigate to the Discovery screen.
You can also use the deep link to test the navigation `ioit://itw/discovery/info`

Preview:
IMAGE 2024-11-06 15:07:35

### CODE REVIEW

File: ts/features/itwallet/navigation/useItwLinkingOptions.tsx [hunk 1]
```diff
@@ -28,6 +29,9 @@ export const useItwLinkingOptions = (): PathConfigMap<AppParamsList> => {
         }),
         ...(canItwBeActivated && {
           [ITW_ROUTES.DISCOVERY.INFO]: "discovery/info"
+        }),
+        ...(itwAlreadyActive && {
+          [ITW_ROUTES.DISCOVERY.ALREADY_ACTIVE_SCREEN]: "discovery/info"
```
Comment (mastro993) [inline ts/features/itwallet/navigation/useItwLinkingOptions.tsx:34]:
```suggestion
        ...(canItwBeActivated && {
          [ITW_ROUTES.DISCOVERY.INFO]: "discovery/info"
        }),
        ...(itwAlreadyActive && {
          [ITW_ROUTES.DISCOVERY.ALREADY_ACTIVE_SCREEN]: "discovery/info"
```

These two routes should be mutually exclusive. What do you think?

Comment (ale-mazz) [reply ts/features/itwallet/navigation/useItwLinkingOptions.tsx:34]: @mastro993 Maybe we could do something like this?

```
  ...(canItwBeActivated
    ? { [ITW_ROUTES.DISCOVERY.INFO]: "discovery/info" }
    : itwAlreadyActive
    ? { [ITW_ROUTES.DISCOVERY.ALREADY_ACTIVE_SCREEN]: "discovery/info" }
    : {})
```

Comment (mastro993) [reply ts/features/itwallet/navigation/useItwLinkingOptions.tsx:34]: Yes, but inverted!

File: ts/features/itwallet/navigation/routes.ts [hunk 2]
```diff
@@ -3,7 +3,8 @@ export const ITW_ROUTES = {
   ONBOARDING: "ITW_CARD_ONBOARDING" as const,
   DISCOVERY: {
     INFO: "ITW_DISCOVERY_INFO",
-    IPZS_PRIVACY: "ITW_IPZS_PRIVACY"
+    IPZS_PRIVACY: "ITW_IPZS_PRIVACY",
+    ALREADY_ACTIVE_SCREEN: "ALREADY_ACTIVE_SCREEN"
```
[the remark this answers is not in the document]
Comment (ale-mazz) [reply ts/features/itwallet/navigation/routes.ts:7]: Sure, thanks for the suggestion!

**Review by mastro993 — APPROVED** [review]
LGTM

File: ts/features/itwallet/navigation/useItwLinkingOptions.tsx (hunk 1 above)
Comment (gispada) [reply ts/features/itwallet/navigation/useItwLinkingOptions.tsx:34]: How about something like this?

```ts
{
  ...(isUserAllowedToItw && {
    [ITW_ROUTES.ISSUANCE.CREDENTIAL_ASYNC_FLOW_CONTINUATION]:
      "credential/issuance",
    [isItwValid
      ? ITW_ROUTES.DISCOVERY.ALREADY_ACTIVE_SCREEN
      : ITW_ROUTES.DISCOVERY.INFO]: "discovery/info"
  })
}
```

**Review by gispada — APPROVED** [review]
LGTM

**Pull request merged**